src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -100,6 +100,12 @@
   @Nullable
   public abstract String getConfigSchemaVersion();
 
+  public boolean isDiscogapic() {
+    return (getTransportProtocol() == TransportProtocol.HTTP
+        && getConfigSchemaVersion() != null
+        && getConfigSchemaVersion().startsWith("1."));
```

Comment:
Add a brief motivating comment for `startsWith("1.")`?

Reply:
Both DIREGAPIC and DISCOGAPIC have `TransportProtocol` as `HTTP`, but discogapic is always gapic yaml v1, while DIREGAPIC is always gapic yaml v2 or no gapic yaml at all. This is the most reliable way to distinguish between the two without introducing a new "transport protocol". Also discogapic and diregapic share significant portion of logic, that is why I'm reusing `TransportProtocol.HTTP` in DIREGAPIC..